src/ITILSolution.php

```diff
@@ -195,6 +195,31 @@ public function prepareInputForAdd($input)
             $this->item->getFromDB($input['items_id']);
         }
 
+        // Check ticket mandatory fields
+        $ticket = $this->item;
+        $tt = $ticket->getITILTemplateToUse(0, $ticket->fields['type'] ?? '', $ticket->fields['itilcategories_id'] ?? '', $ticket->fields['entities_id'] ?? '');
+        if (count($tt->mandatory)) {
+            $mandatory_missing = [];
+            $fieldsname        = $tt->getAllowedFieldsNames(true);
+            foreach ($tt->mandatory as $key => $val) {
+                if (
+                    (empty($ticket->fields[$key]) || ($ticket->fields[$key] == 'NULL'))
+                    || !isset($ticket->fields[$key])
+                ) {
```

Comment:
`!isset($ticket->fields[$key])` is part of `empty($ticket->fields[$key])`.

```suggestion
                if (empty($ticket->fields[$key]) || ($ticket->fields[$key] == 'NULL')) {
```